createProxy.js

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+export default function createProxy(x, setHandler) {
+  const map = new WeakMap();
+  const handler = {
+    get(target, key) {
+      const value = target[key];
+      if(typeof value === 'object' && value !== null) {
+        let proxy = map.get(value);
+        if(!proxy) {
+          proxy = new Proxy(value, handler);
+          map.set(value, proxy);
+        }
+        return proxy;
+      }
+      return value;
+    },
+    set(target, key, value) {
+      if(!target.hasOwnProperty(key)) {
```

Comment:
Should `tarket[key] = value;` occur in either case?  Here's a repl.it to illustrate my question.  https://repl.it/repls/ComposedChocolateApplicationprogrammer

Reply:
It's up to the `setHandler` to set in that case:

https://repl.it/repls/ImmaculateGoldBooleanvalue

Reply:
OK, so it is the case that the expected behavior is that `setHandler` fires only when new properties are added to an object, but not when existing properties are updated (as demonstrated in the REPL)?

Reply:
Yes... and perhaps we should do `{addProperty, setProperty}` options instead ... and allow for them to have defaults.

Reply:
I pushed a change to support two handlers with defaults.